pkg/controllerutil/container_util.go

```diff
@@ -38,14 +38,17 @@ func InjectZeroResourcesLimitsIfEmpty(c *corev1.Container) {
 	if c.Resources.Limits == nil {
 		c.Resources.Limits = corev1.ResourceList{}
 	}
+	if c.Resources.Requests == nil {
+		c.Resources.Requests = corev1.ResourceList{}
+	}
 
-	safeSetLimitValue := func(name corev1.ResourceName) {
-		if _, ok := c.Resources.Requests[name]; !ok {
-			if _, ok = c.Resources.Limits[name]; !ok {
-				c.Resources.Limits[name] = zeroValue
-			}
+	safeResourceValue := func(resource corev1.ResourceList, name corev1.ResourceName) {
+		if _, ok := resource[name]; !ok {
+			resource[name] = zeroValue
 		}
 	}
-	safeSetLimitValue(corev1.ResourceCPU)
-	safeSetLimitValue(corev1.ResourceMemory)
+	safeResourceValue(c.Resources.Requests, corev1.ResourceCPU)
+	safeResourceValue(c.Resources.Requests, corev1.ResourceMemory)
+	safeResourceValue(c.Resources.Limits, corev1.ResourceCPU)
+	safeResourceValue(c.Resources.Limits, corev1.ResourceMemory)
```

Comment:
This logic is problematic; if the user has set a request but hasn't set a limit, the limit should not be set to zero here.